### PR TITLE
Fix mkdocs.py to support https or http schema by PUBLIC_API_URL

### PR DESCRIPTION
**Pull request by qqbuby**
image

### CODE REVIEW

File: readthedocs/doc_builder/backends/mkdocs.py [hunk 1]
```diff
@@ -67,7 +68,8 @@ def append_conf(self, **kwargs):
         # Mkdocs needs a full domain here because it tries to link to local media files
         if not media_url.startswith('http'):
             domain = getattr(settings, 'PRODUCTION_DOMAIN')
-            media_url = 'http://{}{}'.format(domain, media_url)
+            scheme = urlparse(settings.PUBLIC_API_URL).scheme
+            media_url = '{}://{}{}'.format(scheme, domain, media_url)
```
Comment (ericholscher) [inline readthedocs/doc_builder/backends/mkdocs.py]: Can we just use `//` here for a protocol relative URL?

**Comment by humitos** [comment]
@qqbuby hi! This PR seems to be close to be merged. There is just one question to answer and maybe a simple fix. Would you like to continue and finish it? Thanks!

**Comment by davidfischer** [comment]
This PR is for a private RTD setup as the `MEDIA_URL` in production on readthedocs.org is `https://media.readthedocs.org`. There are no mixed media warnings in RTD's production setup.

However, in my local tests, @ericholscher's proposed protocol relative URL works fine.

**Comment by ericholscher** [comment]
Closing this, as I don't believe it's valid.

**Pull request closed without merging**